### PR TITLE
Fix some MPI and host/device memory issues in MDS FR problem

### PR DESCRIPTION
**Pull request by nychiang**
Fix some bugs happened when MDS FR is used on device.
Now MDS FR problem uses the same `compute_mode` and `mem_space` as the base problem.
In addition, change the MPI `comm` in FR problem to the one used by base problem.

This PR solves the `copy_to` problem found in some exago+pridecomp tests.

### CODE REVIEW

**Comment by nychiang** [comment]
BTW, I noticed that function [hiopNlpFormulation::finalizeInitialization](https://github.com/LLNL/hiop/blob/4e3faa9963e83cd7f2a388bfcc1e402dc86fc81d/src/Optimization/hiopNlpFormulation.cpp#L168) uses host memory to set up the problem. Therefore the interface function [interface_base.get_vars_info](https://github.com/LLNL/hiop/blob/4e3faa9963e83cd7f2a388bfcc1e402dc86fc81d/src/Optimization/hiopNlpFormulation.cpp#L220) only uses host memory.

File: src/LinAlg/hiopVectorRajaPar.cpp [hunk 1]
```diff
@@ -1947,9 +1947,16 @@ void hiopVectorRajaPar::set_array_from_to(hiopInterfaceBase::NonlinearityType* a
   if(end - start == 0)
     return;
   
-  auto& rm = umpire::ResourceManager::getInstance();
   hiopInterfaceBase::NonlinearityType* vv = const_cast<hiopInterfaceBase::NonlinearityType*>(arr_src); // <- cast away const
```
Comment (cnpetra) [inline src/LinAlg/hiopVectorRajaPar.cpp:1950]: I think the cast is not needed anymore

**Review by jwang125 — APPROVED** [review]
Work as describe.

File: src/LinAlg/hiopVectorRajaPar.cpp (hunk 1 above)
Comment (nychiang) [reply src/LinAlg/hiopVectorRajaPar.cpp]: Done!

**Review by cnpetra — APPROVED** [review]

**Comment by cnpetra** [comment]
reopening

**Pull request merged**